skaffold/src/test/kotlin/com/google/container/tools/skaffold/run/SkaffoldCommandLineStateTest.kt

```diff
@@ -0,0 +1,135 @@
+
+import com.google.common.truth.Truth.assertThat
+import com.google.container.tools.skaffold.SkaffoldExecutorService
+import com.google.container.tools.skaffold.SkaffoldExecutorSettings
+import com.google.container.tools.skaffold.SkaffoldProcess
+import com.google.container.tools.skaffold.message
+import com.google.container.tools.test.ContainerToolsRule
+import com.google.container.tools.test.TestService
+import com.google.container.tools.test.expectThrows
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.RunnerAndConfigurationSettings
+import com.intellij.execution.configurations.RunConfigurationBase
+import com.intellij.execution.configurations.SearchScopeProvider
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.util.ThrowableRunnable
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+
+/** Unit tests for [SkaffoldCommandLineState] */
+class SkaffoldCommandLineStateTest {
+    @get:Rule
+    val containerToolsRule = ContainerToolsRule(this)
+
+    private lateinit var skaffoldCommandLineState: SkaffoldCommandLineState
+
+    @MockK
+    private lateinit var mockExecutionEnvironment: ExecutionEnvironment
+    @MockK
+    private lateinit var mockRunnerSettings: RunnerAndConfigurationSettings
+    @MockK
+    @TestService
+    private lateinit var mockSkaffoldExecutorService: SkaffoldExecutorService
+
+    private val skaffoldSettingsCapturingSlot: CapturingSlot<SkaffoldExecutorSettings> = slot()
+
+    @Before
+    fun setUp() {
+        every {
+            mockExecutionEnvironment.runnerAndConfigurationSettings
+        } answers { mockRunnerSettings }
+
+        // CommandLineState calls this static method, needs to be mocked
+        mockkStatic(SearchScopeProvider::class)
+        every { SearchScopeProvider.createSearchScope(any(), any()) } answers { mockk() }
+
+        // Skaffold executor answer mocks
+        val mockSkaffoldProcess: SkaffoldProcess = mockk(relaxed = true)
+        every {
+            mockSkaffoldExecutorService.executeSkaffold(capture(skaffoldSettingsCapturingSlot))
+        } answers { mockSkaffoldProcess }
+    }
+
+    @Test
+    fun `unsupported run configuration type throws execution exception`() {
+        every { mockRunnerSettings.configuration } answers { mockk<RunConfigurationBase>() }
+
+        skaffoldCommandLineState = SkaffoldCommandLineState(
+            mockExecutionEnvironment,
+            SkaffoldExecutorSettings.ExecutionMode.DEV
+        )
+
+        val exception = expectThrows(
+            ExecutionException::class,
+            ThrowableRunnable { skaffoldCommandLineState.startProcess() })
+        assertThat(exception.message).isEqualTo(message("skaffold.unsupported.run.settings"))
```

Comment:
instead of using the message from the bundle in the unit test, I think its better to have the message string again in the unit test (this is the pattern we've used in the other plugin too). This way, we also are testing that the message in the bundle matches what we expect.

You should also then be able to get rid of `@VisibleForTesting` on `#message`

Reply:
Sounds good, I thought about that too, and probably we won't be changing the messages a lot, so this shouldn't be a maintenance issue.